packages/bpk-component-datatable/src/BpkDataTable.js

```diff
@@ -25,13 +26,31 @@ import _omit from 'lodash/omit';
 import STYLES from './BpkDataTable.scss';
 import BpkDataTableColumn from './BpkDataTableColumn';
 import hasChildrenOfType from './hasChildrenOfType';
-import makeSorter from './sorter';
+import makeSorter, { type Sorter } from './sorter';
+import type { Props } from './common-types';
 
 const getClassName = cssModules(STYLES);
 const omittedTableProps = ['rowGetter', 'rowCount', 'onHeaderClick'];
 
-class BpkDataTable extends Component {
-  constructor(props) {
+type State<Row> = {
+  sorter: Sorter<Row>,
+  rowSelected: ?number,
+};
+
+class BpkDataTable<Row> extends Component<Props<Row>, State<Row>> {
```

Comment:
Should we have a `type Props` object similar to `type State` where we are checking the props passed that are explicitly called out in the props table https://github.com/Skyscanner/backpack/tree/master/packages/bpk-component-datatable or is it sufficiently covered by the propTypes declaration at the bottom of this file?